nav2_lifecycle_manager/include/nav2_lifecycle_manager/lifecycle_manager.hpp

```diff
@@ -239,6 +239,8 @@ class LifecycleManager : public rclcpp::Node
   rclcpp::TimerBase::SharedPtr bond_timer_;
   rclcpp::TimerBase::SharedPtr bond_respawn_timer_;
   std::chrono::milliseconds bond_timeout_;
+  std::chrono::milliseconds service_timeout_;
+  bool non_default_timeout;
```

Comment:
```suggestion
  bool non_default_timeout_;
```

Reply:
Also, I see a `get_parameter` for this, but no declare

Reply:
"service_timeout" declared above https://github.com/ros-navigation/navigation2/blob/1d64f8ae6673b640d992c433de5d0698493a1e79/nav2_lifecycle_manager/src/lifecycle_manager.cpp#L45

But as you suggested below, we can do it without the legacy handling, so I will probably remove "non_default_timeout"

Reply:
OK!